### PR TITLE
[pickers] Add details in invalid mask error

### PR DESCRIPTION
**Pull request by alexfauquette**
Part of #4487

I customize the error message depending on the main problems encountered in the issue

I'm wondering if we should write a doc page to explain how format and mask can be customized

### CODE REVIEW

File: packages/x-date-pickers/src/internals/utils/text-field-helper.test.ts [hunk 1]
```diff
@@ -66,7 +66,10 @@ describe('text-field-helper', () => {
         expect(runMaskValidation()).to.be.equal(true);
       } else {
         expect(runMaskValidation).toWarnDev(
-          `The mask "${mask}" you passed is not valid for the format used ${format}. Falling down to uncontrolled not-masked input.`,
+          [
+            `The mask "${mask}" you passed is not valid for the format used ${format}.`,
+            `Falling down to uncontrolled not-masked input.`,
```
Comment (flaviendelangle) [inline packages/x-date-pickers/src/internals/utils/text-field-helper.test.ts:71]:
```suggestion
            `Falling down to uncontrolled un-masked input.`,
```

or

```suggestion
            `Falling down to uncontrolled no-mask input.`,
```

File: packages/x-date-pickers/src/internals/utils/text-field-helper.ts [hunk 2]
```diff
@@ -79,9 +79,33 @@ export function checkMaskIsValidForCurrentFormat(
...
   if (!isMaskValid && utils.lib !== 'luxon' && process.env.NODE_ENV !== 'production') {
-    console.warn(
-      `The mask "${mask}" you passed is not valid for the format used ${format}. Falling down to uncontrolled not-masked input.`,
-    );
+    const defaultWarning = [
+      `The mask "${mask}" you passed is not valid for the format used ${format}.`,
+      `Falling down to uncontrolled no-mask input.`,
+    ];
+
+    if (format.includes('MMM')) {
```
Comment (flaviendelangle) [inline packages/x-date-pickers/src/internals/utils/text-field-helper.ts:87]: Not sure to understand why we do this `includes('MMM')` check

Comment (flaviendelangle) [reply packages/x-date-pickers/src/internals/utils/text-field-helper.ts:87]: OK, is this incompatibility impossible to change (in the future) or is it just a quick fix waiting for a rework ?

We are in the `if(utils.lib == 'luxon')` is it intended ?

Comment (alexfauquette) [reply packages/x-date-pickers/src/internals/utils/text-field-helper.ts:87]: It's not impossible. For later we could imagine that typing `'1 j'` proposes to complete with `'1 jun'` `'1 jul'` or `'1 jan'` but not for now.

Seems that luxon library was causing problems. I don't know why they decided to do hide the warning message for this specific library

**Review by flaviendelangle — APPROVED** [review]

**Pull request merged**